internal/controller/streams.go

```diff
@@ -196,21 +198,48 @@ func (sm *streamManager) StreamResources(ctx context.Context, logger *zap.Logger
 		logger.Error("Failed to send cluster metadata", zap.Error(err))
 		return err
 	}
+	var allWatchInfos []watcherInfo
+
+	// PHASE 1: List all resources
 	for resource, apiGroup := range resourceAPIGroupMap {
```

Comment:
Iterate over the resources in a deterministic order? This may make troubleshooting easier?
```suggestion
	for _, resource := slices.Sorted(maps.Keys(resourceAPIGroupMap)) {
		apiGroup := resourceAPIGroupMap[resource]
```